chapi2/chapiclient/chapiclient_linux.go

```diff
@@ -0,0 +1,201 @@
+// (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+
+package chapiclient
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hpe-storage/common-host-libs/connectivity"
+	"github.com/hpe-storage/common-host-libs/linux"
+	log "github.com/hpe-storage/common-host-libs/logger"
+)
+
+// GetSocketName returns unix socket name (per process)
+func GetSocketName() string {
+	var socketName string
+	if IsChapidRunning(ChapidSocketPath + ChapidSocketName) {
+		socketName = ChapidSocketPath + ChapidSocketName
+	} else {
+		socketName = ChapidSocketPath + ChapidSocketName + strconv.Itoa(os.Getpid())
+	}
+	return socketName
+}
+
+// NewChapiSocketClient to create chapi socket client
+func NewChapiSocketClient() (*Client, error) {
+	var chapiClient *Client
+
+	socketName := GetSocketName()
+	log.Traceln("setting up chapi client with socket", socketName)
+	// setup chapi client with timeout
+	socketClient := connectivity.NewSocketClient(socketName)
+	chapiClient = &Client{client: socketClient, socket: socketName}
+	return chapiClient, nil
+}
+
+// NewChapiSocketClientWithTimeout to create chapi socket client with timeout
+func NewChapiSocketClientWithTimeout(timeout time.Duration) (*Client, error) {
+	var chapiClient *Client
+
+	socketName := GetSocketName()
+	log.Traceln("setting up chapi client with socket", socketName, "and timeout", timeout)
+	// setup chapi client with timeout
+	socketClient := connectivity.NewSocketClientWithTimeout(socketName, timeout)
+	chapiClient = &Client{client: socketClient, socket: socketName}
+	return chapiClient, nil
+}
+
+// TODO: Consider removing this function after merging to default
+
+// NewChapiClientWithTimeout returns the chapi client with timeout configuration that communicates over the required unix socket(per process or common service)
+func NewChapiClientWithTimeout(timeout time.Duration) (*Client, error) {
+	var chapiClient *Client
+	var err error
+
+	chapiClient, err = NewChapiSocketClientWithTimeout(timeout)
+	if err != nil {
+		return nil, err
+	}
+	return chapiClient, nil
+}
+
+// GetSerialNumber returns the volume serial number as per platform format
+func GetSerialNumber(serialNumber string) string {
```

Comment:
we were thinking of dropping this logic from client side, so serial number for clients is always volume serial number. and linux CHAPI server will prepend "2" when checking for multipath device.